Expand-app-developers-pages (#777)
buildpacks/docs: content/docs/for-app-developers/how-to/build-inputs/specify-default-launch-process.md

summary="Buildpacks can define multiple processes for an application image. Specify which process should be the default."
+++

> Buildpacks usually define the default process type for an application, while retaining the ability for a user to specify their desired default process.

Author: updated

> To configure the `build` to work differently from the default behavior:

Reviewer: ```suggestion
To configure specify the default process:
```

Author: updated

> * You first need to know what processes would be contributed by the buildpacks running in your build.

Reviewer: ```suggestion
* You first need to know what named process types might be contributed by the buildpacks in your build; for more information, see docs for [running the application][Run the application]
```

Author: updated

> * Once known, you can append the following flag to the `pack build` command

Reviewer: ```suggestion
* Append the following flag to the `pack build` command:
```

Author: updated

```bash

> pack build --default-process <process name> <image name>` # <process name> must be a valid process name in launch.toml

Reviewer: ```suggestion
pack build --default-process <process name> <image name>` # <process name> must be a valid process name
```

Author: updated

```

> If this flag is not provided by the user, `pack` will provide the process type as `web` to the `lifecycle`.

Reviewer: ```suggestion
```

I think this is actually not true for newer Platform API versions. We should probably update the pack help text but that is another story

> >As an app developer, you can specify the default process for an application. However, buildpacks-built images can contain multiple process types, to see how to invoke each one, see the [Run the application] page.

Reviewer: ```suggestion
```
